src/Microsoft.Framework.Runtime/FileGlobbing/PatternsCollectionHelper.cs

```diff
@@ -13,9 +13,24 @@ public static class PatternsCollectionHelper
     {
         private static readonly char[] PatternSeparator = new[] { ';' };
 
-        public static IEnumerable<string> GetPatternsCollection(JObject rawProject, string projectDirectory, string propertyName, string[] defaultPatterns)
+        public static IEnumerable<string> GetPatternsCollection(JObject rawProject, string projectDirectory, string propertyName, IEnumerable<string> defaultPatterns = null)
         {
             var token = rawProject[propertyName];
+
+            try
+            {
+                return GetPatternsCollection(token, projectDirectory, defaultPatterns);
+            }
+            catch (InvalidOperationException innerException)
+            {
+                throw new InvalidOperationException("Project json doesn't contain qualified token for property " + propertyName + ".", innerException);
```

Comment:
The inner exception doesn't have the information regarding which property is wrong. Catch and re-throw with the property name in the message.